osversion.go

```diff
@@ -2,16 +2,34 @@ package rig
 
 // OSVersion host operating system version information
 type OSVersion struct {
-	ID      string
-	IDLike  string
-	Name    string
-	Version string
+	ID          string
+	IDLike      string
+	Name        string
+	Version     string
+	extraFields map[string]string
 }
 
 // String returns a human readable representation of OSVersion
-func (o *OSVersion) String() string {
+func (o OSVersion) String() string {
 	if o.Name != "" {
 		return o.Name
 	}
 	return o.ID + " " + o.Version
 }
+
+// SetExtraField sets an extra field
+func (o *OSVersion) SetExtraField(key, value string) {
+	if o.extraFields == nil {
+		o.extraFields = make(map[string]string)
+	}
+	o.extraFields[key] = value
+}
+
+// GetExtraField retrieves an Extra Field value if it is set
+func (o OSVersion) GetExtraField(key string) (string, bool) {
```

Comment:
`GetExtraField` somehow bothers me because it's a value we're getting, not the field.

I was thinking, what if `extraFields` was `raw`. Then there would be `SetRaw(k, v)` (or `Set`?) and `GetRaw(k) (string, bool)` (or `Get`?`).

The parser would fill out the standard fields but also feed everything to the raw. Then you could do something like `os.GetRaw("ID_LIKE")`.

I don't know if this makes any sense outside of the linux version.

🤔

Reply:
I dropped the accessor patter, as per our discussion, and stuck with the direct exposed string map.